### ISSUE TITLE
`client.resolvePreviewURL()` can actually return `null` if user lacks sufficient coverage of document types link resolving

### ISSUE THREAD

**Issue by lihbr** [prismicio/prismic-client#225]
### Versions

- @prismicio/client: v6 (latest)
- node: v16 (latest LTS)

### Reproduction

I ran into issues where `client.resolvePreviewURL()` actually returns null when the user hasn't provided sufficient link resolving strategies, thus the route resolver and the link resolver weren't able to resolve the previewed document link, ultimately leading to this function to return `null`. This can also happen when one might preview a layout-type document that is not expected to resolve to a link.

In short, the non-null assertion here is a bit of a stretch I think: https://github.com/prismicio/prismic-client/blob/master/src/client.ts#L1221-L1223

### Steps to reproduce

Run `client.resolvePreviewURL()` without using the route resolver options and providing it with a link resolver.

### What is expected?

`client.resolvePreviewURL()` return `args.defaultURL`

### What is actually happening?

`client.resolvePreviewURL()` returns `null`

Thanks @5t3ph for the highlight 🙏

**Comment by angeloashmore** [comment]
This is fixed in `v6.4.0` 🎉

Implementation details are provided in #226
